packages/apps/plugins/plugin-sketch/src/hooks/adapter.ts

```diff
@@ -259,5 +280,58 @@ const decode = (value: any): any => {
   if (typeof value === 'object' && value !== null) {
     return Object.fromEntries(Object.entries(value).map(([key, value]) => [key, decode(value)]));
   }
+
   return value;
 };
+
+/**
+ * Insert records into the store catching possible schema errors.
+ * @returns The new schema version or undefined if no migration was needed.
+ */
+// TODO(burdon): What if sync with peer on different version?
+const maybeMigrateSnapshot = (store: TLStore, records: any[], version?: number): number | void => {
+  try {
+    log.info('loading', { records: records.length, schema: version });
+    store.put(records);
+  } catch (err) {
+    log.info('migrating schema...', err);
+    const serialized = records.reduce<Record<string, any>>((acc, record) => {
+      acc[record.id] = record;
+      return acc;
+    }, {});
+
+    const snapshot = store.migrateSnapshot({ schema: schema[version ?? DEFAULT_VERSION], store: serialized });
+    try {
+      log.info('loading', { records: Object.keys(snapshot.store).length, schema: snapshot.schema.schemaVersion });
+      store.loadSnapshot(snapshot);
+      log.info('migrated schema', { version: CURRENT_VERSION });
+      return CURRENT_VERSION;
+    } catch (err) {
+      // Fallback.
+      // TODO(burdon): Notify user shapes missing.
+      log.info('loading records...');
+      for (const record of Object.values(serialized)) {
+        try {
+          store.put([record]);
+        } catch (err) {
+          log.catch(err, { record });
+        }
+      }
+    }
+  }
+};
+
+// TODO(burdon): Need unit test and check that records are duplicated.
```

Comment:
I don't know enough about the tldraw impl to assess this too much, but I'd say based on our other experiences with migrating data that grabbing some real sketches from existing prod spaces and testing this on them manually would be a good sanity check before shipping this.'

Reply:
grabbing = copy and paste right? that works.
i just noticed though that copying from tldraw.com (which i assume is 2.0) doesn't work.